openbas-api/src/main/java/io/openbas/utils/PayloadMapper.java

```diff
@@ -91,8 +96,18 @@ public StatusPayloadOutput getStatusPayloadOutputFromInject(Optional<Inject> inj
               .arguments(statusPayload.getArguments())
               .prerequisites(statusPayload.getPrerequisites())
               .externalId(statusPayload.getExternalId())
-              .executableFile(statusPayload.getExecutableFile())
-              .fileDropFile(statusPayload.getFileDropFile())
+              .executableFile(
+                  Optional.ofNullable(statusPayload.getExecutableFile())
+                      .orElse(
+                          Optional.ofNullable(statusPayload.getExecutableFileId())
+                              .flatMap(this.documentRepository::findById)
+                              .orElse(null)))
+              .fileDropFile(
+                  Optional.ofNullable(statusPayload.getFileDropFile())
+                      .orElse(
+                          Optional.ofNullable(statusPayload.getFileDropFileId())
+                              .flatMap(this.documentRepository::findById)
```

Comment:
Call to documentRepo findById is done twice. Could we maybe check the optional logic before and only call it once if needed?